src/Microsoft.OData.Core/MultipartMixed/ODataMultipartMixedBatchWriterUtils.cs

```diff
@@ -1,43 +1,45 @@
 //---------------------------------------------------------------------
-// <copyright file="ODataBatchWriterUtils.cs" company="Microsoft">
+// <copyright file="ODataMultipartMixedBatchWriterUtils.cs" company="Microsoft">
 //      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
 // </copyright>
 //---------------------------------------------------------------------
 
-namespace Microsoft.OData
+namespace Microsoft.OData.MultipartMixed
 {
     #region Namespaces
     using System;
+    using System.Collections.Generic;
     using System.Diagnostics;
     using System.Globalization;
     using System.IO;
+    using System.Linq;
     #endregion Namespaces
 
     /// <summary>
     /// Helper methods used by the ODataBatchWriter.
     /// </summary>
-    internal static class ODataBatchWriterUtils
+    internal static class ODataMultipartMixedBatchWriterUtils
     {
         /// <summary>
         /// Creates a new batch boundary string based on a randomly created GUID.
         /// </summary>
-        /// <param name="isResponse">A flag indicating whether the boundary should be created for a request or a resonse.</param>
+        /// <param name="isResponse">A flag indicating whether the boundary should be created for a request or a response.</param>
         /// <returns>The newly created batch boundary as string.</returns>
         internal static string CreateBatchBoundary(bool isResponse)
         {
             string template = isResponse ? ODataConstants.BatchResponseBoundaryTemplate : ODataConstants.BatchRequestBoundaryTemplate;
-            return string.Format(CultureInfo.InvariantCulture, template, Guid.NewGuid().ToString());
+            return string.Format(CultureInfo.InvariantCulture, template, Guid.NewGuid());
         }
 
         /// <summary>
         /// Creates a new changeset boundary string based on a randomly created GUID.
         /// </summary>
-        /// <param name="isResponse">A flag indicating whether the boundary should be created for a request or a resonse.</param>
+        /// <param name="isResponse">A flag indicating whether the boundary should be created for a request or a response.</param>
         /// <returns>The newly created changeset boundary as string.</returns>
         internal static string CreateChangeSetBoundary(bool isResponse)

```

Comment:
>CreateChangeSetBoundary [](start = 31, length = 23)

Pass in the groupid passed to startchangesetimplementation rather than create it here.